### PR TITLE
Compression algo

### LINKED ISSUE #237: Response compression selector
Black box exporter can handle compressed responses, but it needs to know what type of compression is used ahead of time. Currently, all compressed responses are getting ignored. We should put a control in the check editor that allows users to specify the compression response type.

### PR DESCRIPTION
**Pull request by rdubrock**
Closes #237 
Depends on https://github.com/grafana/synthetic-monitoring-agent/issues/124

Adds a selector for HTTP checks to allow decompressing compressed responses.

Screen Shot 2021-05-05 at 10 45 13 AM

### CODE REVIEW

File: src/types.ts [hunk 1]
```diff
@@ -444,3 +447,11 @@ export enum CheckListViewType {
   Card,
   List,
 }
+
+export enum HTTPCompressionAlgo {
+  None = '',
```
Comment (electron0zero) [inline src/types.ts:452]: nit: None is capitalized but rest are not. 
```suggestion
  none = '',
```

ps: nits are trivial comments, feel free to ignore them.

File: src/validation.test.ts [hunk 2]
```diff
@@ -24,6 +33,7 @@ describe('trivial cases', () => {
       basicMetricsOnly: true,
       settings: {
         http: {
+          compression: HTTPCompressionAlgo.None,
```
Comment (electron0zero) [inline src/validation.test.ts:36]:
```suggestion
          compression: HTTPCompressionAlgo.none,
```

File: src/components/CheckEditor/checkFormTransformations.ts [hunk 3]
```diff
@@ -58,6 +60,7 @@ export function fallbackSettings(t: CheckType): Settings {
           method: HttpMethod.GET,
           ipVersion: IpVersion.V4,
           noFollowRedirects: false,
+          compression: HTTPCompressionAlgo.None,
```
Comment (electron0zero) [inline src/components/CheckEditor/checkFormTransformations.ts:63]:
```suggestion
          compression: HTTPCompressionAlgo.none,
```

File: src/components/constants.ts [hunk 4]
```diff
@@ -274,10 +275,19 @@ export const CHECK_LIST_VIEW_TYPE_OPTIONS = [
...
 export const PEM_HEADER = '-----BEGIN CERTIFICATE-----';
 
 export const PEM_FOOTER = '-----END CERTIFICATE-----';
 
 export const CHECK_LIST_VIEW_TYPE_LS_KEY = 'grafana.sm.checklist.viewType';
 
 export const INVALID_WEB_URL_MESSAGE = 'Target must be a valid web URL';
+
+export const HTTP_COMPRESSION_ALGO_OPTIONS = [
+  { label: 'none', value: HTTPCompressionAlgo.None },
```
Comment (electron0zero) [inline src/components/constants.ts:288]:
```suggestion
  { label: 'none', value: HTTPCompressionAlgo.none },
```

**Review by electron0zero — APPROVED** [review]
lgtm :+1:

**Review by mem — APPROVED** [review]

**Pull request merged**